### PR TITLE
[FrameworkBundle] fixes displaying of deprecation notices.

### PR DESCRIPTION
**Pull request by hhamon**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | ~ |
| License | MIT |
| Doc PR | ~ |

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/debug_prod.xml [hunk 1]
```diff
@@ -16,7 +16,7 @@
             <tag name="monolog.logger" channel="php" />
             <argument /><!-- Exception handler -->
             <argument type="service" id="logger" on-invalid="null" />
-            <argument /><!-- Log levels map for enabled error levels -->
+            <argument>null</argument><!-- Log levels map for enabled error levels -->
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/debug_prod.xml]: For reference, here is the line that controls what happens in prod:
https://github.com/hhamon/symfony/blob/71e19de147f9d5c7688a827b2915510f298470eb/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L145

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/logger.html.twig [hunk 2]
```diff
@@ -88,10 +88,13 @@
 
     {% if collector.logs %}
         <ul class="alt">
-            {% for log in collector.logs if priority >= 0 and log.priority >= priority or priority < 0 and log.context.type|default(0) == priority %}
-                <li class="{{ cycle(['odd', 'even'], loop.index) }}{% if log.priority >= 400 %} error{% elseif log.priority >= 300 %} warning{% endif %}{% if log.context.scream is defined %} scream{% endif %}">
-                    {{ logger.display_message(loop.index, log) }}
-                </li>
+            {% for log in collector.logs %}
+                {% set is_deprecation = log.context.level is defined and log.context.type is defined and (constant('E_DEPRECATED') == log.context.type or constant('E_USER_DEPRECATED') == log.context.type) %}
+                {% if priority == 'deprecation' ? is_deprecation : (priority >= 0 and log.priority >= priority and not is_deprecation) %}
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/logger.html.twig]: This new filter excludes deprecations from being listed on first page display.
I checked the behavior on 2.3, and they are always listed, whatever the selected level in the filter.
I propose we keep the 2.3 behavior in 2.6 (thus revet most of this patch in twig+collector), and work on a better UI for the selector on 2.7

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by stof** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @hhamon.

**Pull request merged**